biovectors_modules/word2vec_analysis_helper.py

```diff
@@ -141,26 +173,34 @@ def get_neighbors(
         return []
 
     else:
-        word_vector_row = shared_tokens.index(query_token)
-        sim_mat = squareform(1 - pdist(word_vector_matrix, "cosine"), "tomatrix")
-        token_neighbors = sim_mat[word_vector_row, :].argsort()[-neighbors:][::-1]
+        word_vector_matrix = word_vector_matrix.sort_values("token")
+
+        sim_mat = 1 - cdist(
+            word_vector_matrix.query(f"token == {repr(query_token)}")
+            .drop("token", axis=1)
+            .values,
+            word_vector_matrix.drop("token", axis=1).values,
+            "cosine",
+        )
```

Comment:
Minor: it might be better to factor out the two arrays you are trying to compare into two variables, and then pass them to `cdist`. It's a bit hard to read right now.